apstools/beamtime/manage_ioc.sh

```diff
@@ -0,0 +1,174 @@
+
+DEFAULT_SESSION_NAME=apsbss
+DEFAULT_IOC_PREFIX=ioc:bss:
+
+SHELL_SCRIPT_NAME=${BASH_SOURCE:-${0}}
+SELECTION=${1:-usage}
+SESSION_NAME=${2:-"${DEFAULT_SESSION_NAME}"}
+IOC_PREFIX=${3:-"${DEFAULT_IOC_PREFIX}"}
+
+IOC_BINARY=softIoc
+EPICS_DATABASE=apsbss.db
+START_IOC_COMMAND="${IOC_BINARY} -m P=${IOC_PREFIX} -d ${EPICS_DATABASE}"
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+if [ -z "$IOC_STARTUP_DIR" ] ; then
+    # If no startup dir is specified, use the directory with this script
+    IOC_STARTUP_DIR=$(dirname "${SHELL_SCRIPT_NAME}")
+fi
+
+# echo "SESSION_NAME = ${SESSION_NAME}"
+# echo "IOC_PREFIX = ${IOC_PREFIX}"
+# echo "START_IOC_COMMAND = ${START_IOC_COMMAND}"
+# echo "SHELL_SCRIPT_NAME = ${SHELL_SCRIPT_NAME}"
+# echo "IOC_STARTUP_DIR = ${IOC_STARTUP_DIR}"
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+function checkpid() {
+    MY_UID=$(id -u)
+    # # The '\$' is needed in the pgrep pattern to select vm7, but not vm7.sh
+    IOC_PID=$(pgrep ${IOC_BINARY}\$ -u ${MY_UID})
+    # #!echo "IOC_PID=${IOC_PID}"
+
+    if [ "${IOC_PID}" != "" ] ; then
+        # Assume the IOC is down until proven otherwise
+        IOC_DOWN=1
+        SCREEN_PID=""
+
+        # At least one instance of the IOC binary is running;
+        # Find the binary that is associated with this script/IOC
+        for pid in ${IOC_PID}; do
+            # compare directories
+            BIN_CWD=`readlink /proc/${pid}/cwd`
+            IOC_CWD=`readlink -f ${IOC_STARTUP_DIR}`
+
+            if [ "$BIN_CWD" = "$IOC_CWD" ] ; then
+                # The IOC is running;
+                # the process with PID=$pid is the
+                # IOC that was run from $IOC_STARTUP_DIR
+                P_PID=$(ps -p ${pid} -o ppid=)
+                # strip leading (and trailing) whitespace
+                arr=($P_PID)
+                P_PID=${arr[0]}
+                SCREEN_SESSION="${P_PID}.${SESSION_NAME}"
+                SCREEN_MATCH=$(screen -ls "${SCREEN_SESSION}" | grep "${SESSION_NAME}")
+                if [ "${SCREEN_MATCH}" != "" ] ; then
+                    # IOC is running in screen
+                    IOC_DOWN=0
+                    IOC_PID=${pid}
+                    SCREEN_PID=${P_PID}
+                    break
+                fi
+            fi
+        done
+    else
+        # IOC is not running
+        IOC_DOWN=1
+    fi
+
+    return ${IOC_DOWN}
+}
+
+function console () {
+    if checkpid; then
+        echo "Connecting to ${SCREEN_SESSION}'s screen session"
+        # The -r flag will only connect if no one is attached to the session
+        #!screen -r ${SESSION_NAME}
+        # The -x flag will connect even if someone is attached to the session
+        screen -x ${SCREEN_SESSION}
+    else
+        echo "${SCREEN_NAME} is not running"
+    fi
+}
+
+function exit_if_running() {
+    # ensure that multiple, simultaneous IOCs are not started by this user ID
+    MY_UID=`id -u`
```

Comment:
Codacy found an issue: [Use $(...) notation instead of legacy backticked `...`.](https://app.codacy.com/gh/BCDA-APS/apstools/pullRequest?prid=5764884)